internal/pkg/template/templates/cicd/pipeline_cfn.yml

```diff
@@ -104,6 +104,14 @@ Resources:
               - ecr:CompleteLayerUpload
             Resource: '*'
             Condition: {StringEquals: {'ecr:ResourceTag/copilot-application': {{$.AppName}}}}
+          {{- if eq .Source.ProviderName "CodeCommit" }}
+          {{- if ne .Source.OutputArtifactFormat "" }}
```

Comment:
(here and two places below): I wonder if we should make this more strict, so there's less chance of failures after churning. Maybe we should look for only `CODEBUILD_CLONE_REF` since that's the only option besides the default.

"Values can be either CODEBUILD_CLONE_REF or CODE_ZIP. If unspecified, the default is CODE_ZIP."

Reply:
Discussed offline.

If the value to `OutputArtifactFormat` is invalid, CodePipeline will still deploy successfully (with neither default nor full clone selected), and CloudFormation will NOT error out either. That is, the services will just swallow the invalid input. Us swallowing the invalid input is not really different from them swallowing the input.

We should either faithfully pass the value - either valid or invalid - to CFN, or error out early at manifest validation.

Reply:
That is a good catch! I think we need a better validation for the pipeline manifest.